### DISCUSSION TITLE
Strange behavior of ui.scene because of changing the camera

### DISCUSSION THREAD

**Discussion by v479038280** [zauberzeug/nicegui#3249] [General]
Here is the code:
```py
from nicegui import ui

with ui.scene() as scene:
    scene.extrusion([[0,0],[0,5],[5,5],[2.5,2.5],[5,0]],5)
    scene.move_camera(x=0, y=0, z=10)

ui.run()
```

The expect result should look like this:
![微信截图_20240619222304](https://github.com/zauberzeug/nicegui/assets/25096045/2195aceb-3bea-485e-bb5c-170ef4430c08)

but the default view is this:
![微信截图_20240619222501](https://github.com/zauberzeug/nicegui/assets/25096045/9b2c214b-265f-4bfe-97e7-0435e20928d6)

when the scene is clicked the view become this:
![微信截图_20240619222304](https://github.com/zauberzeug/nicegui/assets/25096045/bacfa357-5556-4a4f-bd55-f97f00f2063b)

when refreshed, it becomes this again:
![微信截图_20240619222501](https://github.com/zauberzeug/nicegui/assets/25096045/97859fc3-856a-45ce-9be8-62940e022d95)

Here is the result:
![GIF 2024-06-19 21-08-28](https://github.com/zauberzeug/nicegui/assets/25096045/ce12f17b-e754-4da8-a1cd-34c93aa9b40f)

**Comment by python-and-novella** [comment]
what's your expect?

**Reply by v479038280** [reply]
I add some description to the problem.The view changed when we click the grid area.

**Comment by falkoschindler** [comment]
Hi @v479038280,

I think there's an ambiguity with your camera orientation. The camera pose is defined by its position, its viewing direction ("look at") and the "up" vector. I don't know [Three.js](https://threejs.org/)' default values by heart, but I assume your up vector (probably `[0,0,1]`) now points into the same direction like or opposite to the viewing direction. This is why the rendered scene flips around undecidedly. Try setting the up vector explicitly:
```py
scene.move_camera(x=0, y=0, z=10, up_x=0, up_y=1, up_z=0)
```

**Reply by v479038280** [reply]
Thank you for your reply,  I haven't kook into other parameters besides x y z,  this will also work `scene.move_camera(z=10)'

**Reply by v479038280** [reply]
It seems I've seen the usage of (x=0, y=0, z=some value) in either Rosys or Field_Friend, and I haven't encountered this kind of exception. I can't recall exactly where it was, I just have a vague memory. I really admire your ability to reference so many similar discussions.
